### ISSUE TITLE
Some problem with running vl_simplenn.m

### ISSUE THREAD

**Issue by huangsjb** [vlfeat/matconvnet#131]
Hi, I have encountered some problem in trying to run vl_simplenn.m to test the model.

I simply run the sample code in the following tutorial:
http://www.vlfeat.org/matconvnet/pretrained/

And the following error message is printed:
" Reference to non-existent field 'filters'.

Error in vl_simplenn (line 153)
      res(i+1).x = vl_nnconv(res(i).x, l.filters, l.biases, 'pad', l.pad, 'stride', l.stride) ;

Error in cnn_test (line 16)
res = vl_simplenn(net, im_) ; "

If someone has encountered the same problem as running the sample code? or have any experience on dealing with it?

Thanks a lot anyway.

**Comment by geyun12** [comment]
I meet the same problem with you... hope someone help us.

**Comment by huangsjb** [comment]
Also, I tried to do prediction using the cnn_mnist.m in the examples folder.

By changing the last layer to "softmax" (as discussed in #34), it works. But the tutorial example problem is not solved yet.

**Comment by geyun12** [comment]
If I used the  imagenet-vgg-f model,then it runs successfully. But when I used the imagenet-vgg-verydeep-16, then I meet the same problem as yours.

**Comment by Jerrynet** [comment]
@huangsjb  you can update to the latest version (1.0-beta11), which supports legacy net structure.
They updated net weight structure from '.filters'/'.biases' to '.weights'
A quick workaround is change the code to this:
```
res(i+1).x = vl_nnconv(res(i).x, l.weights{1}, l.weights{2}, 'pad', l.pad, 'stride', l.stride) ;
```

**Comment by wangchuanxd** [comment]
@huangsjb  Maybe your training network did not parallel for your training data setting. As you said ,you changed the layer to "softmax" and it worked.

**Comment by isalirezag** [comment]
I have the same issue.
Could anyone solve this problem?
I received this error as well:
Error in vl_simplenn (line 178)
        res(i+1).x = vl_nnconv(res(i).x, l.weights{1}, l.weights{2}, 'pad', l.pad, 'stride',
        l.stride) ;

**Comment by Mona77** [comment]
I get the same error as well:

Attempt to execute SCRIPT vl_nnconv as a function:
/home/m/matconvnet-1.0-beta12/matlab/vl_nnconv.m

Error in vl_simplenn (line 178)
        res(i+1).x = vl_nnconv(res(i).x, l.weights{1}, l.weights{2}, 'pad', l.pad, 'stride', l.stride) ;

Error in pre_train_vgg_s (line 15)
res = vl_simplenn(net, im_) ;

**Comment by wangchuanxd** [comment]
Sorry for replying your problem so late. I encountered the problem when I setted the error parameters  beteween two layers. Then, I fixed the parameters and it worked. 



wangchuansnnu@163.com
 
From: Mona Fathollahi
Date: 2015-05-28 12:53
To: vlfeat/matconvnet
CC: wangchuanxd
Subject: Re: [matconvnet] Some problem with running vl_simplenn.m (#131)
I get the same error as well:
Attempt to execute SCRIPT vl_nnconv as a function:
/home/m/matconvnet-1.0-beta12/matlab/vl_nnconv.m
Error in vl_simplenn (line 178)
res(i+1).x = vl_nnconv(res(i).x, l.weights{1}, l.weights{2}, 'pad', l.pad, 'stride', l.stride) ;
Error in pre_train_vgg_s (line 15)
res = vl_simplenn(net, im_) ;

**Comment by arBest** [comment]
I am trying to use it for cifar.
I get the same error as well. I am using the option "legacy net".

Thanks

**Comment by arBest** [comment]
Could some one help out?

**Comment by arBest** [comment]
Compiled the mex files with "vl_compilenn". That works.

**Comment by lenck** [comment]
Hi, `Attempt to execute SCRIPT vl_nnconv as a function:` is because the MatConvNet is not compiled. You should run `vl_compilenn`. 
The error `" Reference to non-existent field 'filters'.` means that your network is of the old format. Use the `vl_simplenn_tidy` which would fix it.
Hope that this would help a bit! :)

**Comment by vedattumen** [comment]
I am trying to use it
I get the same error as well. Error:

Error in vl_simplenn (line 300)
      res(i+1).x = vl_nnconv(res(i).x, l.weights{1}, l.weights{2}, ...

Error in Deneme (line 14)
res = vl_simplenn(net, im_) ;

**Comment by cora9298** [comment]
use 'net.meta.normalization' instead of 'net.normalization'

**Comment by sabirdvd** [comment]
I think this is a bug in the framework when you try to run more than 2 or 3 networks at same time, workspace, clear cmd doesn’t work either, only solution is to restart MATLAB and run the same lines again. What I did, I run the first network then I saved the weight then I restarted  MATLAB, then I run the another one. I tried with this two laptop to confirm this bug.
